shopify-client/src/views/Checkout.vue

```diff
@@ -0,0 +1,191 @@
+
+      <b-form-group
+        description="Enter an address"
+        id="input-group-2"
+        label-for="input-2"
+        label="Address:"
+      >
+        <b-form-input
+          id="input-2"
+          placeholder="Enter address"
+          required
+          v-model="form.address"
+        ></b-form-input>
+      </b-form-group>
+
+      <b-form-group
+        description="Enter an phone number"
+        id="input-group-2"
+        label-for="input-2"
+        label="Phone Number:"
+      >
+        <b-form-input
+          id="input-2"
+          placeholder="Enter phone number"
+          required
+          v-model="form.phoneNumber"
+        ></b-form-input>
+      </b-form-group>
+
+      <div class="mt-5">
+        <b-button class="mr-5" type="submit" variant="primary">Submit</b-button>
+        <b-button type="reset" variant="danger">Reset</b-button>
+      </div>
+    </b-form>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "Checkout",
+  data: function() {
+    return {
+      form: {
+        name: "",
+        address: "",
+        phoneNumber: "",
+      },
+      products: [],
+      show: true,
+      error: "",
+    };
+  },
+  created: function() {
+    this.cart = this.$store.state.cart;
+
+    this.getCartProductData();
+  },
+  methods: {
+    onSubmit(event) {
+      event.preventDefault();
+      try {
+        fetch(`/api/checkout`, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            checkouts: this.products.map((p) => ({
+              id: p.id,
+              quantity: p.quantity,
+            })),
+          }),
+        }).then((response) => {
+          if (response.status !== 200) {
+            this.error = response;
+            return;
+          }
+
+          this.$store.commit("clearCart");
+          this.$router.push("/app/checkout/success");
+        });
+      } catch (error) {
+        this.error = error;
+      }
+    },
+    getCartProductData() {
+      fetch(
+        `/api/products?ids=${Object.keys(this.$store.state.cart)
+          .map(Number)
+          .join(",")}`
+      )
+        .then((response) => response.json())
+        .then((data) => (this.products = data))
+        .catch((error) => {
+          this.error = error;
+        });
+    },
+    onReset(event) {
+      event.preventDefault();
+      // Reset our form values
+      this.form.name = "";
+      this.form.address = "";
+      this.form.phoneNumber = 0;
+      this.error = null;
+      // Trick to reset/clear native browser form validation state
+      this.show = false;
+      this.$nextTick(() => {
+        this.show = true;
+      });
+    },
+    getTotalPrice() {
+      return this.products.reduce(
+        (acc, cur) => acc + cur.quantity * cur.price,
+        0
+      );
+    },
+  },
+};
+</script>
+
+<style scoped>
+.center {
+  margin-left: 500px;
+  margin-right: 500px;
```

Comment:
can you change this to percentage? A large margin would lead to different screen size concerns

Reply:
Margin sizes are not usually done in percentage. Margins are typically specified by breakpoints i.e. `xs`, `sm` etc. Since we haven't done much about responsiveness therefore I just used 500px.

Reply:
alright, then we'll just let it pass
styling is low-priority anyways